Mathlib/Lean/Data/Name.lean

```diff
@@ -0,0 +1,29 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: E.W.Ayers, Floris van Doorn
+-/
+
+import Lean.Data.Name
+
+/-!
+Fixups corresponding to lean4/src/Lean/Data/Name.lean.
+-/
+
+namespace Lean.Name
+
+open Lean
+
+/-- Lean 4 makes declarations which are technically not internal
+(that is, head string does not start with `_`) but which sometimes should
+be treated as such. For example, the `to_additive` attribute needs to
+transform `proof_1` constants generated by `Lean.Meta.mkAuxDefinitionFor`.
+This might be better fixed in core, but until then, this method can act
+as a polyfill.
+Note: this declaration also occurs as `shouldIgnore` in the Lean 4 file `test/lean/run/printDecls`.
+-/
```

Comment:
Can you add a reference to the upstream issue in a comment?

Reply:
Done in https://github.com/leanprover-community/mathlib4/pull/2349/commits/aa1f1844b634bf509934c3b89fb906924b7de1de